### ISSUE TITLE
Add Fortiswitch, FortiAP and Fortiextender

### ISSUE THREAD

**Issue by rootadmin1** [AsBuiltReport/AsBuiltReport.Fortinet.FortiGate#73]
### Description

Please ad the FortiSwitch,AP and Extender config to the report

### Before submitting

- [X] I have read [the documentation](https://www.asbuiltreport.com/user-guide/new-asbuiltconfig), and referred to the [known issues](https://www.asbuiltreport.com/user-guide/known-issues/) before submitting this change request.
- [X] I have checked for previously opened & closed [issues](https://github.com/AsBuiltReport/AsBuiltReport.Fortinet.FortiGate/issues) before submitting this change request.

**Comment by alagoutte** [comment]
Thanks @rootadmin1 for assigned ;)

What do you want to display ? (if you have sample configuration example or better API call ^^)

**Comment by rootadmin1** [comment]
Hi @alagoutte,

I would be nice if we have the fortiswitch/ap/extender config in the documentation. Thinks like name type  firmware version, Ip Address.

I don't have any api calls but below i have some links to the cli documentation.

https://docs.fortinet.com/document/fortigate/7.4.3/cli-reference/207620/config-switch-controller-managed-switch
https://docs.fortinet.com/document/fortigate/7.4.3/cli-reference/39848/wireless-controller

**Comment by alagoutte** [comment]
do you have some configuration example ? (cli) and what do you to send on the report ?

**Comment by msoukhomlinov** [comment]
This could include the following:
1.0 FAP overview listing out the number of FAPs by model and status
1.1 Attached FAPs details grouped by model (models, serials, firmware, operational status, registration status)
2.0 FortiAP profiles config grouped by which are in use
3.0 SSID settings

**Comment by alagoutte** [comment]
> This could include the following: 1.0 FAP overview listing out the number of FAPs by model and status 1.1 Attached FAPs details grouped by model (models, serials, firmware, operational status, registration status) 2.0 FortiAP profiles config grouped by which are in use 3.0 SSID settings

Patches are welcome :)

**Comment by msoukhomlinov** [comment]
Will add this to the todo :)

**Comment by alagoutte** [comment]
and you can make a big patch adding this ;-)

**Comment by kitkat0981** [comment]
is this code still in progress?  There seems to be a lot of sections missing like what is indicated in this discussion.

**Comment by alagoutte** [comment]
> is this code still in progress? There seems to be a lot of sections missing like what is indicated in this discussion.

No, but patches are welcome :)

There is already all function/cmdlet on PowerFGT to get information, only need to work on display

**Comment by kitkat0981** [comment]
Do you mean the ps1 scripts located in the private folder?


Regards,

Nelson

**Comment by alagoutte** [comment]
> Do you mean the ps1 scripts located in the private folder?
> 
> 
> Regards,
> 
> Nelson

yes

**Comment by kitkat0981** [comment]
I can provide the one I built if you want.


Regards,

Nelson

**Comment by alagoutte** [comment]
> I can provide the one I built if you want.
> 
> 
> Regards,
> 
> Nelson

yes, you can ! you need to push a PR with your change, it is an OSS project any people can contribute

**Comment by kitkat0981** [comment]
Not sure how this will work, since first: im not a coder, im using Codex.
Second, I have built a whole web page around your
AsBuiltReport.Fortinet.Fortigate with the ability to load and same
configuration files, etc with the help of Codex. I also added the wireless
section and am actively adding more sections that I see are missing.

I don't mind sharing my whole repo with you and if you want to take some of
the work i have done so far and implement it in your, go right ahead !

My github repo is private but I think I can add you, ill have to check that
out. Again, im not a coder, i just like to build stuff with codex to make
my life easier.

Regards,

Nelson

**Comment by alagoutte** [comment]
Hi Nelson,

Like your prefer, you can share your repo or make some PR with change and if it is ok, i can merge (and no problem if code coming from Codex...)
